### PR TITLE
merge windows buildlogic into main makefile

### PR DESCRIPTION
**Pull request by cgzones**

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -34,13 +40,14 @@ DEFINES+=-DREMUSER=\"${OSSEC_USER_REM}\"
 DEFINES+=-DGROUPGLOBAL=\"${OSSEC_GROUP}\"
 DEFINES+=-DMAILUSER=\"${OSSEC_USER_MAIL}\"
 
-LDFLAGS+=-lm
+OSSEC_LDFLAGS=${LDFLAGS} -lm
```
Comment (jrossi) [inline src/Makefile]: MUCH better plan why of doing this.

Comment (cgzones) [inline src/Makefile]: so you can call make for e.g.:
`make TARGET=server CC=my_gcc CFLAGS=my_flags LDFLAGS=my_flags2`
cause i had the experience that predefined variables can not be modified in a makefile

Comment (jrossi) [inline src/Makefile]: Agreed - much much better - kicking myself for ever writing the way it was before :)

File: src/Makefile [hunk 2]
```diff
@@ -1125,11 +1188,65 @@ test_coverage: build_tests
...
+	${OSSEC_CC} ${OSSEC_CFLAGS} -UOSSECHIDS -DARGV0=\"ossec-win32ui\" -c $^ -o $@
+
+win32/ossec-agent.exe: win32/icon.o win32/win_agent.o win32/win_service.o ${syscheck_o} ${rootcheck_o} $(filter-out client-agent/main.o, $(filter-out client-agent/agentd.o, $(filter-out client-agent/event-forward.o, ${client_agent_o}))) $(filter-out logcollector/main.o, ${os_logcollector_o}) ${os_execd_o} ${ossec_libs} ${ZLIB_LIB}
+	${OSSEC_CCBIN} -DARGV0=\"ossec-agent\" -DOSSECHIDS ${OSSEC_CFLAGS} $^ ${OSSEC_LDFLAGS} -o $@
+
+win32/ossec-agent-eventchannel.exe: win32/icon.o win32/win_agent.o win32/win_service.o ${syscheck_o} ${rootcheck_o} $(filter-out client-agent/main.o, $(filter-out client-agent/agentd.o, $(filter-out client-agent/event-forward.o, ${client_agent_o}))) $(filter-out logcollector/main-event.o, ${os_logcollector_eventchannel_o}) ${os_execd_o} ${ossec_libs} ${ZLIB_LIB}
+	${OSSEC_CCBIN} -DARGV0=\"ossec-agent\" -DOSSECHIDS -DEVENTCHANNEL_SUPPORT ${OSSEC_CFLAGS} $^ ${OSSEC_LDFLAGS} -o $@
+
+win32/ossec-rootcheck.exe: win32/icon.o win32/win_service_rk.o ${rootcheck_rk_o} ${ossec_libs}
+	${OSSEC_CCBIN} -DARGV0=\"ossec-rootcheck\" ${OSSEC_CFLAGS} $^ ${OSSEC_LDFLAGS} -o $@
```
Comment (jrossi) [inline src/Makefile]: As pointed out we can remove this.  Now or later does not matter to me

**Pull request merged**